Shared/Samples/Add custom dynamic entity data source/AddCustomDynamicEntityDataSourceView.Model.swift

```diff
@@ -0,0 +1,140 @@
+                let lines = contents.split(separator: "\n")
+                
+                let decoder = JSONDecoder()
+                
+                // The feed events that the feed will emit.
+                let feedEvents = lines
+                    .compactMap { jsonString -> CustomDynamicEntityFeedEvent? in
+                        guard let decodable = try? decoder.decode(
+                            Vessel.self,
+                            from: jsonString.data(using: .utf8)!
+                        ) else { return nil }
+                        
+                        // The geometry that was decoded from the JSON.
+                        let geometry = decodable.geometry
+                        
+                        // We successfully decoded the vessel JSON so we should
+                        // add that vessel as a new observation.
+                        return .newObservation(
+                            geometry: Point(x: geometry.x, y: geometry.y, spatialReference: .wgs84),
+                            attributes: decodable.attributes.makeDictionary()
+                        )
+                    }
+                
+                return VesselFeed(events: feedEvents)
+            }
+            
+            let customDataSource = CustomDynamicEntityDataSource(info: info) { try makeFeed() }
+            
+            dynamicEntityLayer = DynamicEntityLayer(dataSource: customDataSource)
+            
+            let trackDisplayProperties = dynamicEntityLayer.trackDisplayProperties
+            trackDisplayProperties.showsPreviousObservations = true
+            trackDisplayProperties.showsTrackLine = true
+            trackDisplayProperties.maximumObservations = 20
+            
+            let labelDefinition = LabelDefinition(
+                labelExpression: SimpleLabelExpression(simpleExpression: "[VesselName]"),
+                textSymbol: TextSymbol(color: .red, size: 12)
+            )
+            labelDefinition.placement = .pointAboveCenter
+            
+            dynamicEntityLayer.addLabelDefinition(labelDefinition)
+            dynamicEntityLayer.labelsAreEnabled = true
+            
+            map.addOperationalLayer(dynamicEntityLayer)
+        }
+    }
+}
+
+private extension Array where Element == Field {
+    /// An array of fields that match the attributes of each observation in the data source.
+    ///
+    /// This schema is derived from the first row in the custom data source.
+    static var vesselFields: [Field] = [
+        Field(type: .text, name: "MMSI", alias: "MMSI", length: 256),
+        Field(type: .float64, name: "SOG", alias: "SOG", length: 8),
+        Field(type: .float64, name: "COG", alias: "COG", length: 8),
+        Field(type: .text, name: "VesselName", alias: "VesselName", length: 256),
+        Field(type: .text, name: "CallSign", alias: "CallSign", length: 256),
+        Field(type: .text, name: "globalid", alias: "globalid", length: 256)
+    ]
+}
```

Comment:
This property should be computed. Note: Explicit `return` used to silence linter warning.

```suggestion
private extension Array<Field> {
    /// An array of fields that match the attributes of each observation in the data source.
    ///
    /// This schema is derived from the first row in the custom data source.
    static var vesselFields: Self {
        return [
            Field(type: .text, name: "MMSI", alias: "MMSI", length: 256),
            Field(type: .float64, name: "SOG", alias: "SOG", length: 8),
            Field(type: .float64, name: "COG", alias: "COG", length: 8),
            Field(type: .text, name: "VesselName", alias: "VesselName", length: 256),
            Field(type: .text, name: "CallSign", alias: "CallSign", length: 256),
            Field(type: .text, name: "globalid", alias: "globalid", length: 256)
        ]
    }
}
```